sdk/src/assertions/ingredient.rs

```diff
@@ -520,6 +527,103 @@ impl Ingredient {
+
+        match Store::from_stream(format, &mut stream, &mut validation_log, context) {
+            Ok(store) => {
+                ingredient.with_store(&store, &validation_log)?;
+                let manifest_bytes = store.to_jumbf_internal(0)?;
+                Ok((ingredient, Some(manifest_bytes)))
+            }
+            Err(Error::JumbfNotFound)
+            | Err(Error::ProvenanceMissing)
+            | Err(Error::UnsupportedType) => {
+                // no claims but valid file
+                Ok((ingredient, None))
+            }
+            Err(Error::BadParam(desc)) if desc == *"unrecognized file type" => {
+                Ok((ingredient, None))
+            } // no claims but valid file
+            Err(Error::RemoteManifestUrl(url)) | Err(Error::RemoteManifestFetch(url)) => {
+                let status =
+                    ValidationStatus::new_failure(crate::validation_status::MANIFEST_INACCESSIBLE)
+                        .set_url(url)
+                        .set_explanation("Remote manifest not fetched".to_string());
+                let mut validation_results = ValidationResults::default();
+                validation_results.add_status(status.clone());
+                ingredient.validation_results = Some(validation_results);
+                Ok((ingredient, None))
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Add store information to the ingredient assertion
+    pub(crate) fn with_store(
+        &mut self,
+        store: &Store,
+        validation_log: &StatusTracker,
+    ) -> Result<&Self> {
+        self.validation_results = Some(ValidationResults::from_store(store, validation_log));
+
+        if let Some(claim) = store.provenance_claim() {
+            if self.title.is_none() {
+                self.title = claim.title().cloned();
+            }
+            if self.instance_id.is_none() {
+                self.instance_id = Some(claim.instance_id().to_string());
+            }
+
+            let hashes = store.get_manifest_box_hashes(claim);
+
+            self.active_manifest = Some(HashedUri::new(
+                to_manifest_uri(claim.label()),
+                Some(claim.alg().to_owned()),
+                hashes.manifest_box_hash.as_ref(),
+            ));
+            self.claim_signature = Some(HashedUri::new(
+                to_signature_uri(claim.label()),
+                Some(claim.alg().to_owned()),
+                hashes.signature_box_hash.as_ref(),
+            ));
+
+            let thumbnail = claim.thumbnail();
+            if thumbnail.is_some()
+                && self
+                    .validation_results
+                    .as_ref()
+                    .map(|r| r.validation_state())
+                    != Some(crate::ValidationState::Invalid)
+            {
+                self.thumbnail = thumbnail;
+            } else {
+                // todo:: add thumbnail here
```

Comment:
Do we always want one? Maybe here is a place to pull down settings and check what was wished when adding/loading this ingredient?

Reply:
at any rate, I've kept this private to the crate so I can complete the work on this in a later PR.